### PR TITLE
Updating the test for RegExp.prototype[Symbol.search] on the ES6 tab

### PR DESCRIPTION
**Pull request by afmenez**
According to [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1362379#c1), the specification has changed and the results are wrong.

I have updated the test itself and the results for Firefox 49-55, Chrome 51-59 and Edge 13-14, but I was unable to test for Safari10 and XS6.  If someone can re-run the tests on these and post the results here soon, I'll update the patch accordingly.  Or this can be done in a follow-up patch.

### CODE REVIEW

File: data-es6.js
Comment (ljharb) [inline data-es6.js]: if they need checking, let's just remove them from the list.

**Review by ljharb — CHANGES_REQUESTED** [review]

File: data-es6.js
Comment (afmenez) [reply data-es6.js]: Done.

**Review by ljharb — APPROVED** [review]

**Comment by claudepache** [comment]
Safari (10.1 and TP) does not pass the updated test.

**Comment by afmenez** [comment]
OK, the patch is still valid then.

**Comment by claudepache** [comment]
Because the spec change that invalidated the old test (https://github.com/tc39/ecma262/pull/627) was made after ES6 was published, there should be a note saying that the tested semantics are those of a subsequent edition of ES (ES2017 I presume).

**Comment by afmenez** [comment]
Done, but the note didn't appear on the generated html.  Is there a problem on my patch or is there a bug in the generator script?

File: data-es6.js
Comment (ljharb) [inline data-es6.js:8773]: i believe the notes have to go on a "tr" key inside "res" (or on individual results), as opposed to just top-level here.

Comment (afmenez) [reply data-es6.js:8773]: I believe we do this when the note is about the result and not the test like this one.  There are several instances of notes on top-level ("block-level function declaration", "well-known symbols", "non-strict function semantics", etc).  The trouble is I found no other note associated with a subtest instead of the main test, and that's why I believe there's a bug (or limitation) on the generator.

Comment (ljharb) [reply data-es6.js:8773]: Gotcha. In that case we may need to modify the generator as well.

Comment (afmenez) [reply data-es6.js:8773]: I'll open an issue about this.  Do we have to wait until it is fixed to merge this one, or is it better to have it merged to work in the generator bug?

Comment (ljharb) [reply data-es6.js:8773]: I guess they can go in separately; but the generator enhancement can't go in without a note, so either this goes in first, or they go in in the same PR.

Comment (afmenez) [reply data-es6.js:8773]: I have opened #1098 with a reference to this one.  Hopefully this will be merged soon to allow that one to be fixed.

**Pull request merged**
